src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -1181,6 +1211,53 @@ private void addNewFolder() {
 		}
 	}
 
+	private void renameFolder(IElement selected) {
+		String newName = VOptionPane
+				.showInputDialog(mcreator, L10N.t("workspace.elements.folders.rename.message"), L10N.t("workspace.elements.folders.rename.title"), null, new OptionPaneValidatior() {
+					@Override public ValidationResult validate(JComponent component) {
+						String newName = ((JTextField) component).getText();
+						if (!newName.matches("[A-Za-z0-9._ -]+")) {
+							canRename = false;
+							return new Validator.ValidationResult(ValidationResultType.ERROR,
+									L10N.t("workspace.elements.folders.add.error_letters"));
+						}else {
+							canRename = true;
+						}
+						List<FolderElement> folderElements = mcreator.getWorkspace().getFoldersRoot()
+								.getRecursiveFolderChildren();
+
+						for (FolderElement folderElement : folderElements) {
+							if (folderElement.equals(newName)) {

```

Comment:
Don't compare FolderElement with String, this is not ok. Check for names in same folder only as this check does not make sense.

Reply:
fixed that